packages/ui/feature-builder-right-sidebar/src/lib/input-forms/piece-input-form/piece-input-form.component.ts

```diff
@@ -28,32 +45,27 @@ import {
   take,
   tap,
 } from 'rxjs';
-import {
-  AUTHENTICATION_PROPERTY_NAME,
-  ActionErrorHandlingOptions,
-  ActionType,
-  PopulatedFlow,
-  TriggerType,
-  isNil,
-  spreadIfDefined,
-} from '@activepieces/shared';
-import {
-  ActionBase,
-  PieceMetadataModel,
-  PiecePropertyMap,
-  PropertyType,
-  TriggerBase,
-} from '@activepieces/pieces-framework';
-import { FormControl, UntypedFormBuilder, Validators } from '@angular/forms';
+import { HttpRequestWriterComponent } from '../../http-request-writer/http-request-writer.component';
 import { InputFormCore } from '../input-form-core';
 
 @Component({
   selector: 'app-piece-input-form',
   standalone: true,
-  imports: [CommonModule, UiCommonModule, UiFeatureBuilderFormControlsModule],
+  imports: [
+    CommonModule,
+    UiCommonModule,
+    UiFeatureBuilderFormControlsModule,
+    HttpRequestWriterComponent,
+  ],
   changeDetection: ChangeDetectionStrategy.OnPush,
   template: `
-    @if (deps$ | async; as deps) {
+    @if (deps$ | async; as deps) { @if (deps.currentStep?.type ===
+    ActionType.PIECE && deps.currentStep?.settings.pieceName ===
+    "@activepieces/piece-http") {
```

Comment:
add another condition to make sure the selected action is send request, not return response or when there is no action selected.